zeebe/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/value/adhocsubprocess/AdHocSubProcessInstructionRecord.java

```diff
@@ -12,63 +12,65 @@
 import io.camunda.zeebe.msgpack.property.StringProperty;
 import io.camunda.zeebe.msgpack.value.ValueArray;
 import io.camunda.zeebe.protocol.impl.record.UnifiedRecordValue;
-import io.camunda.zeebe.protocol.record.value.AdHocSubProcessActivityActivationRecordValue;
+import io.camunda.zeebe.protocol.record.value.AdHocSubProcessInstructionRecordValue;
 import io.camunda.zeebe.protocol.record.value.TenantOwned;
 import io.camunda.zeebe.util.buffer.BufferUtil;
 import java.util.List;
 
-public final class AdHocSubProcessActivityActivationRecord extends UnifiedRecordValue
-    implements AdHocSubProcessActivityActivationRecordValue {
+public final class AdHocSubProcessInstructionRecord extends UnifiedRecordValue
+    implements AdHocSubProcessInstructionRecordValue {
 
   private static final String DEFAULT_STRING = "";
 
   private final StringProperty adHocSubProcessInstanceKey =
       new StringProperty("adHocSubProcessInstanceKey", DEFAULT_STRING);
-  private final ArrayProperty<AdHocSubProcessActivityActivationElement> elements =
-      new ArrayProperty<>("elements", AdHocSubProcessActivityActivationElement::new);
+  private final ArrayProperty<AdHocSubProcessActivateElementInstruction> activateElements =
+      new ArrayProperty<>("activateElements", AdHocSubProcessActivateElementInstruction::new);
   private final StringProperty tenantId =
       new StringProperty("tenantId", TenantOwned.DEFAULT_TENANT_IDENTIFIER);
 
-  public AdHocSubProcessActivityActivationRecord() {
+  public AdHocSubProcessInstructionRecord() {
     super(3);
-    declareProperty(adHocSubProcessInstanceKey).declareProperty(elements).declareProperty(tenantId);
+    declareProperty(adHocSubProcessInstanceKey)
+        .declareProperty(activateElements)
+        .declareProperty(tenantId);
   }
 
   @Override
   public String getAdHocSubProcessInstanceKey() {
     return BufferUtil.bufferAsString(adHocSubProcessInstanceKey.getValue());
   }
 
-  public AdHocSubProcessActivityActivationRecord setAdHocSubProcessInstanceKey(
+  public AdHocSubProcessInstructionRecord setAdHocSubProcessInstanceKey(
       final String adHocSubProcessInstanceKey) {
     this.adHocSubProcessInstanceKey.setValue(adHocSubProcessInstanceKey);
     return this;
   }
 
   @Override
-  public List<AdHocSubProcessActivityActivationElementValue> getElements() {
-    return elements.stream()
-        .map(AdHocSubProcessActivityActivationElementValue.class::cast)
+  public List<AdHocSubProcessActivateElementInstructionValue> getActivateElements() {
+    return activateElements.stream()
+        .map(AdHocSubProcessActivateElementInstructionValue.class::cast)
```

Comment:
Note: the change from `elements` to `activateElements` and from `AdHocSubProcessActivityActivationElementValue` to `AdHocSubProcessActivateElementInstructionValue` was not defined in the issue, but I figured it would be more expressive to change this in order to clearly indicate what the list of elements represents. I renamed the class name of the elements to align better with what we use in the process modification record.

Happy to change back to `elements` if we want to keep the change smaller.

Reply:
@maff I believe we should drop this commit as this would be a breaking change. AFAIK the current ad-hoc sub-process already uses this record. This means on our log we could have records that already contain this property. It's encoded in MsgPack, and it will be a String representation of the `AdHocSubProcessInstructionRecord`. This String representation contains the key `elements`. If we now rename this key to `activateElements` it will not be able to deserialize it back to the object.

Regarding the rename of the `AdHocSubProcessActivityActivationElement` to `AdHocSubProcessActivateElementInstruction`. I don't have a preference here. Both names are suitable and describe that the object represents imo.